src/Generator/Library.cs

```diff
@@ -175,10 +175,15 @@ public static void SetClassBindName(this ASTContext context, string className, s
                 @class.Name = name;
         }
 
-        public static void SetClassAsValueType(this ASTContext context, string className)
+        public static void SetClassAsValueType(this ASTContext context, string className, Driver driver = null)
         {
             foreach (var @class in context.FindClass(className))
-                @class.Type = ClassType.ValueType;
+            {
+                if (driver != null && driver.Options.IsCLIGenerator && @class.HasBaseClass && @class.BaseClass.IsRefType)
+                    @class.Type = ClassType.RefType;
+                else
+                    @class.Type = ClassType.ValueType;
+            }
```

Comment:
@ddobrev @tritao 
I am not sure about this! Should we leave it upto the user to decide not to make such a class a value type while using CLI generator or is it fine to take the Driver as an argument?
Also, should I leave the default value or shall I change all the files to reflect the changes?